### DISCUSSION TITLE
Why edge data is optional but node data is not?

### DISCUSSION THREAD

**Discussion by hieutran106** [xyflow/xyflow#4586] [Q&A]
We have an observation made during our recent work with custom Edge's data interface. Our initial expectation was that the edge's data would be mandatory. However, upon further examination, it appears that this is not the case.
For example:
```
type CustomEdge = Edge<{label: string}, "my-edge">;
type CustomNode = Node<{label: string}, "my-node">;

type EdgeData = CustomEdge['data'];
type NodeData = CustomNode['data'];
```
We observe that EdgeData can potentially be undefined, represented as `{label: string;} | undefined`, whereas NodeData remains strictly `{label: string;}`
I hope somebody can help me to understand the reasons for behind those design choices.
Thank you.

**Comment by moklick** [comment]
Hey!

Thanks for bringing this up. We made `data` for edges optional because it's not very common for edges to have a data attribute. If it would be required we would have to add an empty data attribute for all edges.
